sap/rfc/user.py

```diff
@@ -168,11 +227,31 @@ def set_valid_to(self, end_date: Union[str]):
 
         return self
 
+    def set_productive_password(self, productive_password: bool):
+        """Set productive password flag"""
+
+        self._userPasswordManager.set_use_productive_password(productive_password)
+        return self
+
+    def get_username(self) -> str:
+        """Get user's name"""
+
+        return self._username
+
+    def get_user_password_manager(self) -> UserPasswordManager:
+        """Set user's password status"""
+
+        return self._userPasswordManager
+
     def _rfc_params_add_username(self, params):
         add_to_dict_if_not_none(params, 'USERNAME', self._username)
 
     def _rfc_params_add_password(self, params):
-        password = copy_dict_or_new(self._password)
+        if self._userPasswordManager.is_productive_password_requested():
+            password = copy_dict_or_new({'BAPIPWD': self._userPasswordManager.get_dummy_password()})
```

Comment:
What happes if the dummy password matches the productive password? I think you should detect that and raise an exception.

Reply:
Done. When passwords are same, then we are throwing an exception.